src/components/MultiHopTrade/components/RateGasRow.tsx

```diff
@@ -37,48 +37,39 @@ import { clickableLinkSx } from '@/theme/styles'
 type RateGasRowProps = {
   affiliateBps: string | undefined
   buyAssetId: AssetId
-  isDisabled?: boolean
   isLoading?: boolean
   rate: string | undefined
   sellAssetId: AssetId
   swapperName: SwapperName | undefined
   swapSource: SwapSource | undefined
   networkFeeFiatUserCurrency: string | undefined
   deltaPercentage?: string | null
-  onClick?: () => void
   invertRate?: boolean
   noExpand?: boolean
   isOpen?: boolean
 } & PropsWithChildren
 
 const helpersTooltipFlexProps: FlexProps = { flexDirection: 'row-reverse' }
 const rowHover = { bg: 'background.surface.raised.base' }
-const rateHover = {
-  cursor: 'pointer',
-  '.rate': { borderColor: 'text.link' },
-}
 
 export const RateGasRow: FC<RateGasRowProps> = memo(
   ({
     affiliateBps,
     buyAssetId,
     children,
-    isDisabled,
     isLoading,
     rate,
     sellAssetId,
     swapperName,
     swapSource,
     networkFeeFiatUserCurrency,
     deltaPercentage,
-    onClick,
     noExpand,
     invertRate,
   }) => {
     const translate = useTranslate()
     const { isOpen, onToggle } = useDisclosure()
-    const [shouldInvertRate, setShouldInvertRate] = useState<boolean | undefined>(invertRate)
-    const [hasClickedRate, setHasClickedRate] = useState(false)
```

Comment:
maybe-blocking: we might need to discuss this with product but it seems like this PR is removing the auto interted rate when switching assets in limit orders input

Reply:
If the user click at least on time, the prop `invertRate` will not be taken in account anymore which is the properly controlling the rate inverting from the parent components, that was the intent

Reply:
fd460e154612311be2a9768d3515ee21eca2b50a